.github/workflows/tools.yaml

```
@@ -18,3 +18,16 @@ jobs:
         run: |
           npm install
           npm run ci
+      - working-directory: ./tools
+        run: |
+          make clean_and_generate
+      - name: Fail when generate introducing diff
+        run: |
+          git diff --exit-code
+      - name: Make sure that no breaking changes were introduced
+        run: | 
+          make tools v2-lint v2-test
+
+
+
+        
```

super nit but let's remove the extra empty lines
```suggestion
      - name: Make sure that no breaking changes were introduced
        run: | 
          make tools v2-lint v2-test
```